automation/functions/delivery.py

```diff
@@ -0,0 +1,174 @@
+from python_http_client.exceptions import BadRequestsError
+from requests.exceptions import HTTPError
+from sendgrid.helpers.mail import Mail
+import structlog
+from structlog.contextvars import bind_contextvars
+
+from automation.utils.templates import render
+
+log = structlog.get_logger("send_delivery_email")
+
+# Turn this off for debugging
+SEND_MAIL = True
+
+
+def on_assigned(
+    ticket,
+    *,
+    member_table,
+    inventory,
+    sendgrid_client,
+    from_email,
+    reply_to=None,
+):
+    bind_contextvars(ticket_id=ticket.ticket_id)
+    log.info("Sending delivery email")
+
+    problem = check_ready_to_send(ticket)
+    if problem:
+        msg = "Ticket not ready to send"
+        log.error(msg, problem=problem)
+        return
+
+    delivery_volunteers = get_delivery_volunteers(ticket, member_table)
+    email = render_email_template(ticket, delivery_volunteers, inventory)
+    email.from_email = from_email
+    if reply_to:
+        email.add_cc(reply_to)
+        email.reply_to = reply_to
+
+    if not SEND_MAIL:
+        log.info(
+            "Would send email", to=email.get()["personalizations"][0]["to"]
+        )
+        return
+    try:
+        sendgrid_client.send(email)
```

Comment:
@leifwalsh Highly encourage enabling Sandbox mode by default for email sends, especially as I just saw we use prod data on staging.  Sandbox can be disabled in prod via the config file.

https://sendgrid.com/docs/for-developers/sending-email/sandbox-mode/
https://github.com/sendgrid/sendgrid-python/blob/main/examples/helpers/mail_example.py#L38

Reply:
Oh sandbox mode sounds great!

Reply:
I am controlling this with an environment variable for now, I'll learn how to do sandbox mode soon.